### PR TITLE
 Fix issues with config `SetURL`

### LINKED ISSUE #149: Default config url causes issues on Windows
This is the output on Windows:
knoxite_v0.0.0-next_Windows_x86_64>knoxite.exe repo init -r C:\Temp\knoxite -p test1243
2020/08/28 00:48:09 error reading the config file: configuration backend 'c' not supported

### PR DESCRIPTION
**Pull request by penguwin**
This pr resolves #149

### CODE REVIEW

**Review by muesli — DISMISSED** [review]

File: config/config_test.go [hunk 1]
```diff
@@ -52,9 +51,24 @@ func TestNew(t *testing.T) {
 		t.Error("Backend for 'mem:' should be a MemoryBackend")
 	}
 
+	_, err = New("~/foobar")
+	if err != nil {
+		t.Fatalf("Failed here: %v", err)
+	}
+
 	_, err = New("c:\\foobar")
```
Comment (ma-hartma) [inline config/config_test.go:59]: According to https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats we could add some more test cases.

File: utils/utils.go [hunk 2]
```diff
@@ -147,3 +149,42 @@ func EncryptionText(enum int) string {
...
+
+func PathToUrl(u string) (*url.URL, error) {
+	url := &url.URL{}
+	// Check if the given string starts with a protocol scheme. Prepend the file
+	// scheme in case none is provided
+	if !isUrl(u) {
+		url.Scheme = "file"
+		url.Path = u
+	} else {
+		// u = url.QueryEscape(u)
```
Comment (ma-hartma) [inline utils/utils.go:169]: This comment may be unintentional

File: utils/utils.go [hunk 3]
```diff
@@ -147,3 +149,42 @@ func EncryptionText(enum int) string {
...
+		var err error
+		url, err = url.Parse(u)
+		if err != nil {
```
Comment (ma-hartma) [inline utils/utils.go:172]: You could just use "return url, err"

**Review by ma-hartma — CHANGES_REQUESTED** [review]

**Comment by ma-hartma** [comment]
Thanks for fixing #151

File: utils/utils.go (hunk 3 above)
Comment (muesli) [reply utils/utils.go:172]: No, we only want to return in case of an error here. Otherwise we'll proceed with the parsed url.

Comment (ma-hartma) [reply utils/utils.go:172]: Oops! Thank you & sorry.

**Review by muesli — APPROVED** [review]

**Review by ma-hartma — APPROVED** [review]

**Pull request merged**